src/MoBi.Presentation/Presenter/SelectReferenceAtParameterValuePresenter.cs

```diff
@@ -56,6 +57,44 @@ public override IEnumerable<ObjectBaseDTO> GetChildObjects(ObjectBaseDTO dto)
          return children;
       }
 
+      public override IReadOnlyList<ObjectPath> GetAllSelections()
+      {
+         var selectedItems = GetAllSelected<IEntity>().ToList();
+
+         var selectedDummyDtos = _view.AllSelectedDTOs
+            .OfType<DummyParameterDTO>()
+            .ToList();
+
+         return replaceMoleculePropertiesWithMoleculeName(selectedItems, selectedDummyDtos);
+      }
+
+      private IReadOnlyList<ObjectPath> replaceMoleculePropertiesWithMoleculeName(
+         IReadOnlyList<IEntity> selectedItems,
+         IReadOnlyList<DummyParameterDTO> selectedParameterDtos)
+      {
+         var newItems = new List<ObjectPath>();
+         foreach (var item in selectedItems)
+         {
+            var temAsPath = CreatePathFor(item);
```

Comment:
@benjaperez1983 Looks good.
Just a typo (I assume `temAsPath `was supposed to be `itemAsPath `:)